eldev.el

```diff
@@ -4028,6 +4031,221 @@ be passed to Emacs, else it will most likely fail."
         :die-on-error  "child Emacs"
         (eldev--forward-process-output "Output of the child Emacs process:" "Child Emacs process produced no output")))))
 
+;; eldev emacs-docker
+
+(defvar eldev--emacs-gui-args
+  (list "-e" "DISPLAY" "-v" "/tmp/.X11-unix:/tmp/.X11-unix")
+  "Arguments needed to launch dockerized Emacs as a GUI.")
+
+(defvar eldev-docker-run-extra-args nil
+  "Extra arguments to pass to \"docker run\".")
+
+(defvar eldev--xhost-hint
+  "It appears your X server is not accepting connections from the docker container.  Have you ran \"xhost +local:root\"? Error trace:\n"
+  "Message to output if it appears the user has not enabled X forwarding.")
+
+(defvar eldev--container-bootstrap-cmd-fn
+  #'eldev--container-bootstrap-cmd-fn
+  "Function to determine the command used by \"docker run\".
+
+It should take one parameter: the arguments of the \"eldev\" call.")
+
+(defvar eldev--docker-home-name "docker-home"
+  "Name of the home directory of the docker user.")
+
+(defvar eldev--docker-os-error-fmt-string
+  "OS %s is not currently supported by \"eldev docker\""
+  "Error message format string if the os is not supported.")
+
+(defun eldev--container-bootstrap-cmd-fn (eldev-args)
```

Comment:
Can we replace bootstrapping with copying the script file (it is now always bundled with Eldev, so simple `locate-file` call should do)? While this sort of implies that we depend on extact target's OS, it's not like bootstrapping commands are cross-platform either. As an added benefit, we'd have "test the source code, not something published" here too, even if for bootstrap scripts it rarely matters.

Reply:
:+1: should be done in the new commit.